xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/XClassDeclaredTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+
+  * See the NOTICE file distributed with this work for additional
+  * information regarding copyright ownership.
+  *
+  * This is free software; you can redistribute it and/or modify it
+  * under the terms of the GNU Lesser General Public License as
+  * published by the Free Software Foundation; either version 2.1 of
+  * the License, or (at your option) any later version.
+  *
+  * This software is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  * Lesser General Public License for more details.
+  *
+  * You should have received a copy of the GNU Lesser General Public
+  * License along with this software; if not, write to the Free
+  * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+
+ */
+
+package org.xwiki.test.ui;
+
+import org.junit.Test;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.test.ui.po.ViewPage;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Check that all mandatory document initializers are properly declared in the demo package build and that the pages
+ * are not created at first start only. See also XWIKI-22368.
+ *
+ * @version $Id$
+ * @since 16.7.0RC1
+ */
+public class XClassDeclaredTest extends AbstractTest
+{
+    private static final String GROOVY_CODE =
+    """
+        {{groovy}}
+        import com.xpn.xwiki.doc.MandatoryDocumentInitializer;
+        import java.time.temporal.ChronoUnit;
+        import java.text.SimpleDateFormat;
+        
+        // define the pages that shouldn't be created during the build.
+        def allowedExceptions = ["XWiki.XWikiServerXwiki"];
```

Comment:
I guess this list should ideally be a Maven/system property.

Reply:
I won't go there for now.